amivapi/auth/auth.py

```diff
@@ -211,15 +210,13 @@ def authenticate(*args):
     # Set defaults
     g.current_token = g.current_session = g.current_user = None
 
-    # Get token
-    token = getattr(request.authorization, 'username', None)
-
-    # Code copied from Eve's TokenAuth - parse different header formats
-    if not token and request.headers.get('Authorization'):
-        token = request.headers.get('Authorization').strip()
-        if token.lower().startswith(('token', 'bearer')):
-            token = token.split(' ')[1]
-    # End of code copied from Eve
+    # Get token: First try basicauth username, else just auth header
+    token = (getattr(request.authorization, 'username', '') or
+             request.headers.get('Authorization', '').strip())
```

Comment:
Somehow this line worries me. This value is basically user controlled and given to the database in line 226.
Are we sure only a simple string will come out of here? Otherwise there might be a way to at least bypass authentication.

Reply:
As far as I know, a http header can *only* be a string

Reply:
I looked into it a bit and it seems to be the case.